src/library/autodj/autodjfeature.cpp

```diff
@@ -298,32 +263,13 @@ void AutoDJFeature::slotAddRandomTrack(bool) {
 }
 
 void AutoDJFeature::constructCrateChildModel() {
-    // Create a crate table-model with a list of crates that have been added
-    // to the auto-DJ queue (and are visible).
-    QSqlTableModel crateListTableModel(this, m_pTrackCollection->getDatabase());
-    crateListTableModel.setTable(CRATE_TABLE);
-    crateListTableModel.setSort(crateListTableModel.fieldIndex(CRATETABLE_NAME),
-                                Qt::AscendingOrder);
-    crateListTableModel.setFilter(CRATETABLE_AUTODJ_SOURCE + " = 1 AND " + CRATETABLE_SHOW + " = 1");
-    crateListTableModel.select();
-    while (crateListTableModel.canFetchMore()) {
-        crateListTableModel.fetchMore();
-    }
-
-    QSqlRecord tableModelRecord = crateListTableModel.record();
-    int nameColumn = tableModelRecord.indexOf(CRATETABLE_NAME);
-    int idColumn = tableModelRecord.indexOf(CRATETABLE_ID);
-
-    // Create a tree-item for each auto-DJ crate.
-    for (int row = 0; row < crateListTableModel.rowCount(); ++row) {
-        int id = crateListTableModel.data(
-            crateListTableModel.index(row, idColumn)).toInt();
-        QString name = crateListTableModel.data(
-            crateListTableModel.index(row, nameColumn)).toString();
-        m_crateList.append(qMakePair(id, name));
-
+    m_crateList.clear();
+    CrateSelectIterator autoDjCrates(m_pTrackCollection->crates().selectAutoDjCrates(true));
+    Crate crate;
+    while (autoDjCrates.readNext(&crate)) {
```

Comment:
It would be cool if this can be a range based loop. 
It looks like we are almost there and just have to add a begin() and an end() function. 
But I might be wrong since I have not done it myself.